File: developer/Samples/NET 4.6/WF_WCF_Samples/WF/Basic/Migration/SimpleInterop/VB/SimpleInterop/WriteLine.vb
'----------------------------------------------------------------
' Copyright (c) Microsoft Corporation.  All rights reserved.
'----------------------------------------------------------------

Imports Microsoft.VisualBasic
Imports System
Imports System.Workflow.ComponentModel

Namespace Microsoft.Samples.WF.Migration
    Public Class WriteLine
        Inherits Activity

        Public Sub New()
            MyBase.New()
        End Sub

        Public Shared ReadOnly TextProperty As DependencyProperty = DependencyProperty.Register("Text", GetType(String), GetType(WriteLine))

        Public Property Text() As String
            Get
                Return CStr(MyBase.GetValue(TextProperty))
            End Get
            Set(ByVal value As String)
                MyBase.SetValue(TextProperty, value)
            End Set
        End Property

        Protected Overrides Function Execute(ByVal context As ActivityExecutionContext) As ActivityExecutionStatus
            Console.WriteLine(Me.Text)
            Return ActivityExecutionStatus.Closed
        End Function
    End Class
End Namespace

File: developer/Samples/NET 4.6/WF_WCF_Samples/WF/Basic/Migration/SimpleInterop/CS/SimpleInterop/WriteLine.cs
//----------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
//----------------------------------------------------------------

using System;
using System.Workflow.ComponentModel;

namespace Microsoft.Samples.WF.Migration
{
    public class WriteLine : Activity
    {
        public WriteLine()
            : base() { }

        public static readonly DependencyProperty TextProperty =
            DependencyProperty.Register("Text", typeof(string), typeof(WriteLine));

        public string Text
        {
            get
            {
                return (string) base.GetValue(TextProperty);
            }
            set
            {
                base.SetValue(TextProperty, value);
            }
        }

        protected override ActivityExecutionStatus Execute(ActivityExecutionContext context)
        {
            Console.WriteLine(this.Text);
            return ActivityExecutionStatus.Closed;
        }
    }
}
